plugins/kubernetes/app/models/kubernetes/client_config_file.rb

```diff
@@ -39,7 +39,7 @@ def parse_file
 
     def parse_clusters
       @clusters = {}
-      @config_file.fetch(:clusters).each do |cluster_hash|
+      (@config_file[:clusters] || []).each do |cluster_hash|
```

Comment:
It will fail validation when trying to add a cluster with that config file, because `Kubernetes::Cluster#test_client_connection` will fail.  But it will show up as a validation error, rather than a 500